esp-hal/src/gpio/mod.rs

```diff
@@ -434,7 +434,11 @@ pub trait Pin: Sealed {
 /// Trait implemented by pins which can be used as inputs.
 pub trait InputPin: Pin {
     #[doc(hidden)]
-    fn waker(&self) -> &'static AtomicWaker;
+    #[inline(always)]
+    fn waker(&self) -> &'static AtomicWaker {
+        static WAKER: AtomicWaker = AtomicWaker::new();
```

Comment:
One waker for all input pins? Or is the `inline(always)` doing something fun with the static?